app/src/main/java/com/framgia/englishconversation/data/source/remote/comment/CommentRemoteDataSource.java

```diff
@@ -140,31 +135,32 @@ public void onChildChanged(DataSnapshot dataSnapshot, String s) {
                         comment.setModifiedAt(
                                 Utils.generateOppositeNumber(comment.getModifiedAt()));
                         comment.setId(dataSnapshot.getKey());
-                        e.onNext(comment);
+                        HashMap<Integer, Comment> commentHashMap = new HashMap<>();
```

Comment:
WARNING: [UseSparseArrays] Use `new SparseArray<Comment>(...)` instead for better performance
 For maps where the keys are of type integer, it's typically more efficient to use the Android `SparseArray` API. This check identifies scenarios where you might want to consider using `SparseArray` instead of `HashMap` for better performance.

This is **particularly** useful when the value types are primitives like ints, where you can use `SparseIntArray` and avoid auto-boxing the values from `int` to `Integer`.

If you need to construct a `HashMap` because you need to call an API outside of your control which requires a `Map`, you can suppress this warning using for example the `@SuppressLint` annotation.